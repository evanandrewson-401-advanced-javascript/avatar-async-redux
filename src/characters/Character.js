import React from 'react';
import { } from 'react-redux';
import PropTypes from 'prop-types';

const Character = ({ _id, name, photoUrl }) => {
  return (
    <>
      <img src={photoUrl} />
      <h1>{name}</h1>
    </>
  );
};

Character.propTypes = {
  toDetailPage: PropTypes.func.isRequired,
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired
};

export default Character;
