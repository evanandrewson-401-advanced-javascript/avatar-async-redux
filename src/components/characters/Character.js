import React from 'react';
import { Link } from 'react-redux';
import PropTypes from 'prop-types';

const Character = ({ _id, name, photoUrl }) => {
  return (
    <Link to={`/character/${_id}`}>
      <img src={photoUrl} />
      <h1>{name}</h1>
    </ Link>
  );
};

Character.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired
};

export default Character;
