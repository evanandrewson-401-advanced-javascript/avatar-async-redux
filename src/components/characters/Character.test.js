import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

const character = {
  _id: '5cf5679a915ecad153ab68c8',
  allies: [
    'Royal Earthbender Guards'
  ],
  enemies: [
    'Chin'
  ],
  photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
  name: '46th Earth King',
  affiliation: ' Earth Kingdom Earth Kingdom Royal Family',
};

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character { ...character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
