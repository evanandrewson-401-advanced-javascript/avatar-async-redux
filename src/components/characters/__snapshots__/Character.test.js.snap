// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<Link
  to="/character/5cf5679a915ecad153ab68c8"
>
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
  />
  <h1>
    46th Earth King
  </h1>
</Link>
`;
